src/mock_adapter/controllers/product.py

```diff
@@ -3,11 +3,19 @@
 
 import appier
 
+import models
+
 class ProductController(appier.Controller):
 
-    @appier.route("/products/<id>.png", "GET")
+    @appier.route("/products/<int:id>.json", "GET")
+    def show(self, id):
+        product = models.Product.get(id = id, map = True)
+        return product
+
+    @appier.route("/products/<int:id>.png", "GET")
     def image(self, id):
-        file = open("resources/images/bikini1.png", "rb")
+        path = "resources/images/product%s.png" % id

```

Comment:
%s works, but you know it's an int, so format like this: "resources/images/product%d.png"